Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs

```diff
@@ -0,0 +1,184 @@
+using Cysharp.Threading.Tasks;
+using Studio23.SS2.SaveSystem.Data;
+using Studio23.SS2.SaveSystem.Interfaces;
+using Studio23.SS2.SaveSystem.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+
+namespace Studio23.SS2.SaveSystem.Core
+{
+
+    [RequireComponent(typeof(FileProcessor))]
+    internal class SaveSlotProcessor : MonoBehaviour
+    {
+
+        [SerializeField] internal SaveSlot _selectedSlot;
+
+        [SerializeField] internal SlotConfiguration _slotConfiguration;
+        [SerializeField] internal FileProcessor _fileProcessor;
+        [SerializeField] internal ArchiverBase archiverBase;
+
+        private async void Start()
+        {
+            _fileProcessor = GetComponent<FileProcessor>();
+            CreateSlotFolders();
+            await SelectSlot(0);
+        }
+
+        internal async UniTask SelectSlot(int index)
+        {
+            _selectedSlot= await GetSaveSlotMetaData(index);
+        }
+
+        internal string GetSelectedSlotPath()
+        {
+            return Path.Combine(_slotConfiguration.SavePathRoot, _selectedSlot.Name);
+        }
+
+        internal void CreateSlotFolders()
+        {
+
+            for (int i = 0; i < _slotConfiguration._slotCount; i++)
+            {
+                SaveSlot slot = new SaveSlot($"Save Slot {i}");
+
+                string slotPath = Path.Combine(_slotConfiguration.SavePathRoot, slot.Name,_slotConfiguration._slotDatafolderName);
+                if (Directory.Exists(slotPath))
+                {
+                    continue;
+                }
+                Directory.CreateDirectory(slotPath);
+
+            }
+        }
+
+        internal async UniTask<SaveSlot> GetSaveSlotMetaData(int index)
+        {
+            SaveSlot slotMeta=new SaveSlot($"Save Slot {index}");
+
+            string slotMetaPath = Path.Combine(_slotConfiguration.SavePathRoot, $"Save Slot {index}", _slotConfiguration._slotMetafileName);
```

Comment:
$"Save Slot {index}"
white space may be problematic on different platforms. it's better to remove white space from the path or folder name

Reply:
Fixed